dChatFilter/dChatFilter.h

```diff
@@ -20,17 +20,17 @@ class dChatFilter
 	dChatFilter(const std::string& filepath, bool dontGenerateDCF);
 	~dChatFilter();
 
-	void ReadWordlistPlaintext(const std::string & filepath);
-	bool ReadWordlistDCF(const std::string & filepath);
-	void ExportWordlistToDCF(const std::string & filepath);
-	bool IsSentenceOkay(const std::string& message, int gmLevel);
+	void ReadWordlistPlaintext(const std::string& filepath, bool whiteList);
+	bool ReadWordlistDCF(const std::string& filepath, bool whiteList);
+	void ExportWordlistToDCF(const std::string& filepath, bool whiteList);
+	std::vector<std::string> IsSentenceOkay(const std::string& message, int gmLevel, bool whiteList = true);
 
 private:
 	bool m_DontGenerateDCF;
-	std::vector<size_t> m_Words;
+	std::vector<size_t> m_NoNoWords;
+	std::vector<size_t> m_YesYesWords;
```

Comment:
I get the funny nature of the lists but this will just cause confusion for people looking at these variables in the future.  I recommend changing them to be m_ApprovedWords and m_RejectedWords

Reply:
Done